solutions/bronze/usaco-739.mdx

```diff
@@ -339,53 +339,48 @@ public class BovineGenomics {
 </JavaSection>
 <PySection>
 
-<Warning>
-
-Due to Python's slow speed, the below code TLEs on a couple test cases.
-
-</Warning>
-
 ```py
-GENOME_ID = {"A": 0, "T": 1, "C": 2, "G": 3}
-
-with open("cownomics.in") as read:
-	cow_num, genome_len = [int(i) for i in read.readline().split()]
-
-	spotted = []
-	for _ in range(cow_num):
-		genome_str = read.readline()
-		genome = []
-		for g in range(genome_len):
-			# A -> 0, C -> 1, T -> 2, G -> 3
-			genome.append(GENOME_ID[genome_str[g]])
-		spotted.append(genome)
-
-	plain = []
-	for _ in range(cow_num):
-		genome_str = read.readline()
-		genome = []
-		for g in range(genome_len):
-			genome.append(GENOME_ID[genome_str[g]])
-		plain.append(genome)
-
-valid_sets = 0
-for a in range(genome_len):
-	for b in range(a + 1, genome_len):
-		for c in range(b + 1, genome_len):
-			spotted_ids = [False for _ in range(64)]
-			for sc in range(cow_num):
-				total = spotted[sc][a] * 16 + spotted[sc][b] * 4 + spotted[sc][c] * 1
-				spotted_ids[total] = True
-
-			for pc in range(cow_num):
-				total = plain[pc][a] * 16 + plain[pc][b] * 4 + plain[pc][c] * 1
-				if spotted_ids[total]:
+# The major takeaway here is that tuples are hashable! Meaning we can check for them in sets in constant time!
+# This code passes all tests!
+# A particular kind of set of 3 genes will gives us a gene-tuple.
+# Any gene-tuple should only be in either spotted or plain cows, else we wont be able to come to a comclusion.
```

Comment:
```suggestion
```
afaik this is the same solution as the one before - just with tuples instead of hashing

Reply:
would you mind putting an info section mentioning how using tuples speeds up the code on python?

syntax:
```
<Info title="">
The hashing used in the C++/Java solutions runs too slow for Python. However, we can use ... instead to pass the time limit.
</Info>
```

Reply:
I think it probably isn't actually faster in general, but it happens to be faster on some test cases because it breaks differently

maybe we should just keep the original solution?

Reply:
👍 or a spoiler with this solution would also be fine imo